tests/test-ptrace.c

```diff
@@ -178,7 +178,7 @@ main (int argc, char **argv)
 
   if (argc == 1)
     {
-      static char *args[] = { "self", "/bin/ls", "/usr", NULL };
+      static char *args[] = { "self", "/usr/bin/env", "ls", "/usr", NULL };
```

Comment:
You're adding more values to the array, you should fix the rest of the code that thinks it knows the size of `args` (like the `fprintf` call below).

Possibly consider using `execvp` down there, avoids the need to specify `/usr/bin/env` or `/bin/ls` (can just call `ls`) and to pass the global `environ` to `execve`.

Reply:
Unless that `self` part is another binary? Haven't analyzed the entirety of the code. Then you might need to use `env`, yeah.

Reply:
Yes, I agree. It's much simpler that way.